lib/ble_manager.dart

```diff
@@ -147,7 +151,7 @@ abstract class BleManager {
   /// On Android [peripheralId] scanned on one  device may or may not be
   /// recognized on a different Android device depending on peripheral’s
   /// implementation and changes in future OS releases.
-  Peripheral createUnsafePeripheral(String peripheralId, {String name});
+  Peripheral createUnsafePeripheral(String peripheralId, {String? name});
```

Comment:
Does an optional argument has to be explicitly typed as nullable? There's no point in explicitly passing a null to an optional argument if that's not the case.

Reply:
yes, that's what the new dart standard says, doesn't even let you "compile" it without it. 
So to sum up name needs to eighter:
- have a default value {String name = ""} (I'm down with that too)
- or be explicitly nullable, so if the caller omits it. it will be null by default implicitly, that needs to be nullchecked thus the type must be marked nullable in that case